src/python/pants/backend/python/goals/run_pex_binary.py

```diff
@@ -66,22 +71,16 @@ async def create_pex_binary_run_request(
                 # variables are not stripped.
                 "--no-strip-pex-env",
             ),
-        ),
-    )
-    requirements_pex_get = Get(
-        Pex,
-        RequirementsPexRequest(
-            addresses=[field_set.address],
-            internal_only=True,
-            resolve_and_lockfile=field_set.resolve.resolve_and_lockfile(python_setup),
+            # From RequirementsPexRequest.
+            additional_inputs=requirements_pex.additional_inputs,
+            requirements=requirements_pex.requirements,
+            sources=requirements_pex.sources,
```

Comment:
Oh.. this negates the whole cache thing that motivated #13693 in the first place, right?
So perhaps a better way to "merge" the pex:es is called for...

Reply:
Or is the caching used properly for the digests we got there.. ?